CI failed with mypy check error

**Describe the bug**

Github Actions CI test failed with following log in master branch

```
  mypy run-test-pre: PYTHONHASHSEED='1504402015'
  mypy run-test: commands[0] | mypy py7zr
  py7zr/helpers.py:93: error: Unused "type: ignore" comment
  py7zr/py7zr.py:1120:49: error: Argument 1 to "_check_7zfile" of "SevenZipFile" has incompatible type "<subclass of "Path" and "IOBase">"; expected "Union[BinaryIO, BufferedReader]"
  py7zr/py7zr.py:1198:46: error: Argument 1 to "open" has incompatible type "Optional[Any]"; expected "Union[Union[str, bytes, PathLike[str], PathLike[bytes]], int]"
  Found 3 errors in 2 files (checked 13 source files)
```

**Related issue**
(if exist)


**Expected behavior**
Should pass

**Condition**
Version: v0.17.0 and after

--- a/py7zr/helpers.py
+++ b/py7zr/helpers.py
@@ -90,7 +90,7 @@ def _calculate_key2(password: bytes, cycles: int, salt: bytes, digest: str):
         for i, c in enumerate(salt + password):
             buf.saltpassword[i] = c
         buf.round = 0
-        mv = memoryview(buf)  # type: ignore # noqa
+        mv = memoryview(buf)
         while buf.round < rounds:
             m.update(mv)
             buf.round += 1

--- a/py7zr/py7zr.py
+++ b/py7zr/py7zr.py
@@ -757,7 +757,7 @@ class SevenZipFile(contextlib.AbstractContextManager):
             self.header_encryption = False
 
     @staticmethod
-    def _check_7zfile(fp: Union[BinaryIO, io.BufferedReader]) -> bool:
+    def _check_7zfile(fp: Union[BinaryIO, io.BufferedReader, io.IOBase]) -> bool:
         result = MAGIC_7Z == fp.read(len(MAGIC_7Z))[: len(MAGIC_7Z)]
         fp.seek(-len(MAGIC_7Z), 1)
         return result
@@ -1116,8 +1116,10 @@ def is_7zfile(file: Union[BinaryIO, str, pathlib.Path]) -> bool:
     """
     result = False
     try:
-        if isinstance(file, io.IOBase) and hasattr(file, "read"):
-            result = SevenZipFile._check_7zfile(file)  # noqa
+        if (isinstance(file, BinaryIO) or isinstance(file, io.BufferedReader) or isinstance(file, io.IOBase)) and hasattr(
+            file, "read"
+        ):
+            result = SevenZipFile._check_7zfile(file)
         elif isinstance(file, str):
             with open(file, "rb") as fp:
                 result = SevenZipFile._check_7zfile(fp)
@@ -1194,7 +1196,9 @@ class Worker:
                             skip_notarget=skip_notarget,
                         )
                 else:
-                    filename = getattr(fp, "name", None)
+                    if getattr(fp, "name", None) is None:
+                        raise InternalError("Caught unknown variable status error")
+                    filename: str = getattr(fp, "name", "")  # do not become "" but it is for type check.
                     self.extract_single(open(filename, "rb"), empty_files, 0, 0, q)
                     concurrent_tasks = []
                     exc_q: queue.Queue = queue.Queue()

